### PR TITLE
Server Framework

### PR DESCRIPTION
**Pull request by Dejaku51**
## About The Pull Request
This PR makes it so servers (currently RnD and tcomms) generate heat and work with efficiency depending on their temperature

Also adds cooling to kilo, rad and meta tcomms

Makes RnD servers generate less heat and use less electricity with better parts

## Why It's Good For The Game
No more 4 part tcomms in random room without proper setup, cooling tcomms actually has sense now

## Testing Photographs and Procedure

1. Test if tcomms work roundstart (they do)
2. turn off freezer in tcomms
3. say different things
4. as the temperature in tcomms rises, efficiency of machines lowers and messages get sent with lag depending on efficiency of the machines handling the message
<details>
<summary>Screenshots&Videos</summary>
Lower efficiency because the server is hot

image

image

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog
:cl:
refactor: Tcomms and RnD servers now generate heat and work with varying efficiency based on the temperature it is
tweak: Added cooling to tcomms in metastation
tweak: Added cooling to tcomms in kilostation
tweak: Added cooling to tcomms in radstation
tweak: RnD servers now use less power and generate less heat if they got better parts
code: changed typepaths of tcomms machines from /obj/machinery/telecomms to /obj/machinery/server/telecomms
code: Added connect_roundstart to atmos connector that will make it automatically connect to any portable atmos machine (ex. canister) on the same tile when spawned
/:cl:

### CODE REVIEW

**Comment by Dejaku51** [comment]
TODO: convert RnD server to this

File: code/__DEFINES/stat.dm [hunk 1]
```diff
@@ -12,10 +12,12 @@
 #define MAX_SATIETY 600
 
 // bitflags for machine stat variable
-#define BROKEN	(1<<0)
-#define NOPOWER	(1<<1)
-#define MAINT	(1<<2)		// under maintaince
-#define EMPED	(1<<3)		// temporary broken by EMP pulse
+#define BROKEN		(1<<0)
+#define NOPOWER		(1<<1)
+#define MAINT		(1<<2)		// under maintaince
```
Comment (EvilDragonfiend) [inline code/__DEFINES/stat.dm]: since you're here fix the typo lol

Comment (Dejaku51) [reply code/__DEFINES/stat.dm]: fixed

**Review by Miliviu — CHANGES_REQUESTED** [review]
Hi! I'm not a maintainer but I've checked the code and LGTM!

But I have a concern! I was testing the RnD Server Power consumption (And it works)
<details>
Before cool parts:

image

After cool parts:

image

</details>

But I couldn't upgrade the Servers, the only way it was to "destroy" the server and build it with new parts

<details>
Here, you can see I'm trying to deconstruct (or just open the latch with the screwdriver a turned off server and I can't 

image
</details>

I'm not sure if this intended, if it is then I'd not see any issues with the PR :)

**Comment by BriggsIDP** [comment]
image
looks much nicer, tyty

**Comment by Dejaku51** [comment]
mean rnd servers in the commit guh

**Review by Miliviu — APPROVED** [review]
Yup! That should do it, LGTM!

**Review by itsmeow — CHANGES_REQUESTED** [review]
Please add an UpdatePaths script for the new servers. (in `tools/UpdatePaths/Scripts`)

**Comment by Dejaku51** [comment]
git is fuck

**Pull request closed without merging**